### PR TITLE
rpcUrl as required and removed redundant code

### PR DESCRIPTION
**Pull request by VGabriel45**
# Summary

In order to fix current and potential new issues when using third party providers like Privy or Particle Auth we will refactor rpcUrl as required field on smart account class instance.

Related Issue: # (issue number)

## Change Type

- [x] Bug Fix
- [x] Refactor
- [ ] New Feature
- [x] Breaking Change
- [x] Documentation Update
- [ ] Performance Improvement
- [ ] Other

# Checklist

- [x] My code follows this project's style guidelines
- [x] I've reviewed my own code
- [x] I've added comments for any hard-to-understand areas
- [ ] I've updated the documentation if necessary
- [x] My changes generate no new warnings
- [x] I've added tests that prove my fix is effective or my feature works
- [x] All unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published

### CODE REVIEW

**Comment by joepegler** [comment]
As discussed we should keep the chainId from the signer and compare it with the chainId from the bundler

**Review by joepegler — CHANGES_REQUESTED** [review]

File: packages/account/tests/account.e2e.spec.ts [hunk 1]
```diff
@@ -30,21 +30,25 @@ describe("Account Tests", () => {
     const smartAccount = await createSmartAccountClient({
       signer,
       bundlerUrl,
+      rpcUrl: mumbai.viemChain.rpcUrls.default.http[0],
```
[the remark this answers is not in the document]
Comment (VGabriel45) [reply packages/account/tests/account.e2e.spec.ts:33]: Ok not sure what's the issue here ? Should not use viemChain instead of mumbai.viemChain ?

**Comment by joepegler** [comment]
I think you can move the failing unit tests into the e2e tests

**Comment by livingrockrises** [comment]
resolve conflicts

**Review by livingrockrises — APPROVED** [review]

**Review by joepegler — APPROVED** [review]

**Pull request merged**